scalafix-core/shared/src/main/scala/scalafix/internal/rule/AsInstanceOfDisabled.scala

```diff
@@ -0,0 +1,31 @@
+package scalafix.internal.rule
+
+import scala.meta._
+import scalafix.rule.SemanticRule
+import scalafix.util.SemanticdbIndex
+import scalafix.rule.RuleCtx
+import scalafix.lint.LintMessage
+import scalafix.lint.LintCategory
+import scalafix.syntax._
+
+case class AsInstanceOfDisabled(index: SemanticdbIndex)
+    extends SemanticRule(index, "AsInstanceOfDisabled")
+    with Product {
+
+  val errorCategory: LintCategory = LintCategory.error(
+    """asInstanceOf is unsafe in isolation and violates parametricity when guarded by isInstanceOf.""".stripMargin
+  )
+
+  override def check(ctx: RuleCtx): Seq[LintMessage] =
+    ctx.tree.collect {
```

Comment:
`ctx.index.names.collect { ... }` will do the same with the benefit of avoiding the tricky `Term.Apply...` match. `SymbolMatcher.normalized` can also be used as a replacement of `s.symbol.contains` with the benefit of being able to support multiple symbols.

Reply:
Changed, thanks for the help.